From e1cda962316a23f14e1b9b33e4d4800b31ded958 Mon Sep 17 00:00:00 2001
From: Constantin Musca <constantin.musca@intel.com>
Date: Fri, 6 Feb 2015 14:36:51 +0200
Subject: [PATCH] inv_mpu6050: Pass the inv_mpu6050_state trig to
 iio_trigger_notify_done

The iio_trigger_notify_done function can be called with a NULL trig
when the user space clears the trigger and this causes a kernel panic.
Use the trig member of the inv_mpu6050_state structure which doesn't
change.

Tracked-On: https://jira01.devtools.intel.com/browse/IRDA-3611
Change-Id: Ifd6befc408311fe2f2316f74dcbc6ef136728765
Signed-off-by: Constantin Musca <constantin.musca@intel.com>
---
 drivers/iio/imu/inv_mpu6050/inv_mpu_ring.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- linux.orig/drivers/iio/imu/inv_mpu6050/inv_mpu_ring.c
+++ linux/drivers/iio/imu/inv_mpu6050/inv_mpu_ring.c
@@ -180,7 +180,7 @@ irqreturn_t inv_mpu6050_read_fifo(int ir
 
 end_session:
 	mutex_unlock(&indio_dev->mlock);
-	iio_trigger_notify_done(indio_dev->trig);
+	iio_trigger_notify_done(st->trig);
 
 	return IRQ_HANDLED;
 
@@ -188,7 +188,7 @@ flush_fifo:
 	/* Flush HW and SW FIFOs. */
 	inv_reset_fifo(indio_dev);
 	mutex_unlock(&indio_dev->mlock);
-	iio_trigger_notify_done(indio_dev->trig);
+	iio_trigger_notify_done(st->trig);
 
 	return IRQ_HANDLED;
 }
